drivers/include/periph/qspi.h

```diff
@@ -0,0 +1,234 @@
+ *           the first falling SCK edge.
+ * - MODE_3: CPOL=1, CPHA=1 - The first data bit is sampled by the receiver on
+ *           the second falling SCK edge.
+ */
+#ifndef HAVE_QSPI_MODE_T
+typedef enum {
+    QSPI_MODE_0 = 0,        /**< CPOL=0, CPHA=0 */
+    QSPI_MODE_1,            /**< CPOL=0, CPHA=1 */
+    QSPI_MODE_2,            /**< CPOL=1, CPHA=0 */
+    QSPI_MODE_3             /**< CPOL=1, CPHA=1 */
+} qspi_mode_t;
+#endif
+
+/**
+ * @brief   Standard QSPI erase block sizes
+ */
+#ifndef HAVE_QSPI_ERASE_SIZE_T
+typedef enum {
+    QSPI_ERASE_4K   = SFLASH_CMD_ERASE_SECTOR,
+    QSPI_ERASE_64K  = SFLASH_CMD_ERASE_BLOCK,
+    QSPI_ERASE_CHIP = SFLASH_CMD_ERASE_CHIP,
+} qspi_erase_size_t;
+#endif
+
+/**
+ * @brief   Configuration Options for QSPI
+ * @{
+ */
+#define QSPI_FLAG_1BIT          (0x0)   /**< single data line           */
+#define QSPI_FLAG_2BIT          (0x1)   /**< two parallel data lines    */
+#define QSPI_FLAG_4BIT          (0x2)   /**< four parallel data lines   */
+#define QSPI_FLAG_8BIT          (0x3)   /**< eight parallel data lines  */
+
+#define QSPI_FLAG_BIT_MASK      (0x3)   /**< mask to get data lines from flags */
+#define QSPI_FLAG_BIT(flags)    ((flags) & QSPI_FLAG_BIT_MASK)
+
+#define QSPI_FLAG_DDR           (0x4)   /**< use Double Data Rate mode  */
+
+#define QSPI_FLAG_ADDR_32BIT    (0x8)   /**< use 32 bit addressing      */
+/** @} */
+
+/**
+ * @brief   Basic initialization of the given QSPI bus
+ *
+ * @note    This function MUST not be called more than once per bus!
+ *
+ * @param[in] bus       QSPI device to initialize
+ */
+void qspi_init(qspi_t bus);
+
+/**
+ * @brief   Configure the QSPI peripheral settings
+ *
+ * @param[in] bus       QSPI device to configure
+ * @param[in] mode      QSPI mode @see qspi_mode_t
+ * @param[in] flags     QSPI Configuration Options
+ * @param[in] clk_hz    QSPI frequency in Hz
+ */
+void qspi_configure(qspi_t bus, qspi_mode_t mode, uint32_t flags, uint32_t clk_hz);
+
+/**
+ * @brief   Start a new QSPI transaction
+ *
+ * Starting a new QSPI transaction will get exclusive access to the QSPI bus
+ * and configure it according to the given values. If another QSPI transaction
+ * is active when this function is called, this function will block until the
+ * other transaction is complete (qspi_relase was called).
+ *
+ * @param[in] bus       QSPI device to access
+ */
+void qspi_acquire(qspi_t bus);
+
+/**
+ * @brief   Finish an ongoing QSPI transaction by releasing the given QSPI bus
+ *
+ * After release, the given SPI bus should be fully powered down until acquired
+ * again.
+ *
+ * @param[in] bus       QSPI device to release
+ */
+void qspi_release(qspi_t bus);
+
+/**
+ * @brief   Execute a command with response data e.g Read Status, Read JEDEC
+ *
+ * @param[in]  bus      QSPI device
+ * @param[in]  command  The JEDEC command ID
+ * @param[out] response Command response data, may be NULL
+ * @param[in]  len      Size of the command response buffer
```

Comment:
Do we have a realistic limit for this? The nRF52 has a 18 bit length register for the read and write lengths, or about 256KiB. Of course I can handle the split in the driver, just wondering if it is necessary to handle it in the driver.

Reply:
Nah we should just return the number of bytes read.
For commands I don't expect large buffers anyway. (and they have to be in memory *somewhere*)

Reply:
Of course, makes sense.